ui/pages/confirmations/components/confirm/info/hooks/use-eip-1559-fees.ts

```diff
@@ -0,0 +1,24 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useEffect, useState } from 'react';
+import { hexToDecimal } from '../../../../../../../shared/modules/conversion.utils';
+
+export const useEIP1559TxFees = (currentConfirmation: TransactionMeta) => {
+  const [maxFeePerGas, setMaxFeePerGas] = useState(0);
+  const [maxPriorityFeePerGas, setMaxPriorityFeePerGas] = useState(0);
+
+  useEffect(() => {
```

Comment:
We're currently using `useState` for the two values plus a `useEffect` to update them whenever the confirmation changes, but we're returning a fresh object each time meaning we're not gaining any caching other than avoiding the tiny cost of repeatedly parsing the values from hex to a number.

Would it be simpler to just always parse and return the values directly without any need for `useState` or `useEffect`?

Or if we want to be super performance conscious, could we instead just use `useMemo` which feels more native as it's for caching expensive operations.

Such as:

```
export const useEIP1559TxFees = (currentConfirmation: TransactionMeta) => {
    return useMemo(() => {
        const maxFeePerGas = ...
        const maxPriorityFeePerGas = ...

        return { maxFeePerGas, maxPriorityFeePerGas };
    }, [currentConfirmation]);
};